File: Golang/MergeTwoSortedLists/MergeTwoSortedLists_test.go
package MergeTwoSortedLists

import (
	"testing"
	"reflect"
)

func TestMergeTwoLists(t *testing.T) {

	node124 := &ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
			Next: &ListNode{
				Val: 4,
			},
		},
	}
	node134 := &ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 3,
			Next: &ListNode{
				Val: 4,
			},
		},
	}

	if reflect.DeepEqual(mergeTwoLists(node124, node134), &ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 1,
			Next: &ListNode{
				Val: 2,
				Next: &ListNode{
					Val: 3,
					Next: &ListNode{
						Val: 4,
						Next: &ListNode{
							Val: 4,
						},
					},
				},
			},
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(mergeTwoLists(node124, nil), node124) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(mergeTwoLists(nil, node134), node134) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if mergeTwoLists(nil, nil) == nil {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/MergeTwoSortedListsTest.py
from unittest import TestCase

from MergeTwoSortedLists import MergeTwoSortedLists, ListNode


class TestMergeTwoSortedLists(TestCase):
    def test_mergeTwoLists(self):
        m = MergeTwoSortedLists()

        n1 = ListNode(1)
        n1.next = ListNode(2)
        n1.next.next = ListNode(4)

        n2 = ListNode(1)
        n2.next = ListNode(3)
        n2.next.next = ListNode(4)

        tmp = ListNode(1)
        tmp.next = ListNode(1)
        tmp.next.next = ListNode(2)
        tmp.next.next.next = ListNode(3)
        tmp.next.next.next.next = ListNode(4)
        tmp.next.next.next.next.next = ListNode(4)
        self.assertEqual(m.mergeTwoLists(n1, n2), tmp)

        tmp = ListNode(1)
        tmp.next = ListNode(2)
        tmp.next.next = ListNode(4)
        self.assertEqual(m.mergeTwoLists(n1, None), tmp)

        tmp = ListNode(1)
        tmp.next = ListNode(3)
        tmp.next.next = ListNode(4)
        self.assertEqual(m.mergeTwoLists(None, n2), tmp)

        self.assertIsNone(m.mergeTwoLists(None, None))
